docs/_concepts/archive-authz.md

```diff
@@ -0,0 +1,96 @@
+
+This concept includes 2 possible solutions. Once a solution is chosen, the concept will be updated accordingly.
+
+## Requirements
+
+This concept aims to satisfy the following requirements:
+
+1. Archiving applications can read from all event types, without exception. They cannot write nor administer event types, unless specified for individual event types;
+2. Event type owners should not be able to accidentally forbid the archiving applications from reading events in an event type;
+3. Event type owners who do not wish for their events to be read by the archiving applications should get in touch with the administrators of the archiving applications;
+4. Event type owners should be made aware that the events published to their event type will be archived;
+5. Only administrators can update the list of archiving applications. This operation should be simple.
+
+The ability to exclude some event types from archival is not discussed in this concept.
+
+## Solution 1: add all archiving applications to the list of authorized event types
+
+The most simple solution is to add the archiving applications to the list of authorized event types:
+
+- If an event type does not have an authorization section, do nothing;
+- If an event type has an authorization section, add the list of archiving applications to the read section;
+- If an event type is updated and has an authorization section, make sure that the archiving applications cannot be removed from the read section.
+
+Nakadi will have to know the list of archiving applications; to do this, a new endpoint will be created:
+```
+POST /settings/resources/archivers
+```
+and its complement:
+```
+GET /settings/resources/archivers
+```
+
+These endpoints will be accessible to administrators with ADMIN rights only.
+
+If the list of authorized applications is updated, all event types with an authorization section will be updated accordingly
+
+### List of Tickets (to be created)
+
+- Create new endpoints
+- Update all event types with authorization when the list of archivers is updated;
+- Add archiving applications to new event types, if there is an authorization section;
+- Forbid updates of event types that remove archiving applications.
+
+### Pros
+
+- Relatively easy to implement: no changes to the authorization logic;
+- Makes it obvious which applications can read from the event type.
+
+### Cons
+
+- Confusing for users: they are not able to remove some applications from the readers, and there is nothing that marks them as archiving applications;
+- Updating the list of archiving applications could be cumbersome, as it may require updating a lot of event types.
+
+## Solution 2: a new resource for archiving applications
+
+In this solution, we create a new resource, `archive`, for which applications can only get READ permissions (write and admin must be empty).
```

Comment:
I don't like the 'archive' word as it will be then 'audit', or 'monitor' etc. I see the need for 'roles' here. We already have regular users, admins, power readers. Maybe we can unify them somehow. Like extending /admins endpoint to accept readers, writers, admins, and data_readers lists.
At least 'archive' should be like 'data_access' or 'global_readers' to not point to the function("why") but to the access pattern ("what").

Reply:
I don't know, for me the current Lionel's suggestion is pretty clear and it doesn't break the general idea that there are resources and for each resource there are three possible types of permission: read/write/admin. Introducing of new permission type "data_readers" will look weird.

Reply:
@v-stepanov yes, my idea was to introduce some kind of auth role-based approach. It can be done extending access levels or extending access to particular resources. I like Lionel's idea to put all roles/resources in one "folder" so it can be extended later if needed. The only problem is that "resources" name does not suggest anything about access control.

Reply:
@SergKam @v-stepanov I'm ok with renaming ('what' is indeed better than 'why' in our current authorization model), but I'm not sure about extending admins. Admins has 3 lists of permissions (read, write, admin), and something like 'global_readers' isn't really a permission.
What could work would be `all_data_access`, for example.

Another option, as @SergKam suggests, is to go for a RBAC model, but that would require more changes. We would have to define roles, and assign permissions to roles (a permission being the right to perform an operation on an object), and assign users (both employees and applications) to roles. In this case, we'd have something like:
`/settings/authz/roles/admin_read`
`/settings/authz/roles/admin_write`
`/settings/authz/roles/admin_all_event_types`
`/settings/authz/roles/all_data_access`
etc.
Current, per-event type authorization could be considered as individual roles, so for each protected event type (eg `etXXX`) there would be 3 roles: `etXXX_read`, `etXXX_write`, `etXXX_admin`. This is where the RBAC model breaks down, as there will be a lot of roles.

I think it is quite a bit more work, and I'm not convinced that we need that yet.

Reply:
But we will have a broken RBAC anyway but like:
`/settings/authz/admins/(read, write, admin)`
`/settings/authz/all_data_access/(read, write, admin)`
it doesn't look intuitive for me as it is our own reinvented wheel.